File: Ecommerce/app/src/main/java/pe/isil/ecommerce/data/remote/ApiClient.java
package pe.isil.ecommerce.data.remote;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class ApiClient {

    private static String BASE_URL="https://short-cut-chair.glitch.me/";

    private static Retrofit retrofit = null;

    public static Retrofit getRetrofit(){
        if (retrofit == null){
            retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit;
    }

}

File: SuperHeroCompose/app/src/main/java/pe/isil/superherocompose/data/remote/ApiClient.kt
package pe.isil.superherocompose.data.remote

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object ApiClient {

    private const val BASE_URL = "https://www.superheroapi.com/api.php/"

    private var retrofit: Retrofit? = null


    private fun getRetrofit(): Retrofit {
        if (retrofit == null) {
            retrofit = Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .build()

        }
        return retrofit as Retrofit
    }

    fun getHeroService(): HeroService {
        return getRetrofit().create(HeroService::class.java)
    }
}